src/utils/queue.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const debug = require('debug')('fastboot:ember-app');
+
+/**
+ * Utility Queue class to store queue of sandboxes that can be leveraged when using `buildSandboxPerVisit`.
+ *
+ * @public
+ */
+class Queue {
+  constructor(maxSize = 1) {
+    this.items = [];
+    this.maxSize = maxSize;
+  }
+
+  enqueue(item) {
+    if (this.size() < this.maxSize) {
+      this.items.push(item);
+    } else {
+      debug('Ignoring adding appInstance to queue as Queue is already full!');
+    }
+  }
+
+  dequeue() {
+    if (!this.isEmpty()) {
+      return this.items.shift();
+    }
+
+    return null;
```

Comment:
Is `null` allowed to be an item in the queue, if so, then maybe document that `null` means either the item is null or the queue is empty. Or you can just disallow nulls somehow.

Reply:
Sure I'll document `null` and what it means here.